diff --git a/src/preload/backtrace-helper.h b/src/preload/backtrace-helper.h
index 1ceac1b..32338dc 100644
--- a/src/preload/backtrace-helper.h
+++ b/src/preload/backtrace-helper.h
@@ -34,6 +34,28 @@ backtrace_init (void)
 #endif
 }
 
+#ifndef unw_backtrace
+int unw_backtrace(void **buffer, int size) {
+    unw_context_t context;
+    unw_cursor_t cursor;
+    if (unw_getcontext(&context) || unw_init_local(&cursor, &context)) {
+        return 0;
+    }
+
+    unw_word_t ip;
+    int current = 0;
+    while (unw_step(&cursor) > 0) {
+        if (current >= size || unw_get_reg(&cursor, UNW_REG_IP, &ip)) {
+            break;
+        }
+
+        buffer[current++] = (void *)(uintptr_t)ip;
+    }
+
+    return current;
+}
+#endif
+
 static int
 backtrace_func (SysprofCaptureAddress  *addrs,
                 guint                   n_addrs,
